phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * Installer's AJAX frontend handler
+ */
+
+(function($) { // Avoid conflicts with other libraries
+    // Global variables
+    var pollTimer = null;
+    var nextReadPosition = 0;
+
+    // Template related variables
+    var contentWrapper = $('.install-body').find('.main');
+
+    // Intercept form submits
+    intercept_form_submit($('#install_install'));
+
+    function poll_content(xhReq)
+    {
+        var messages = xhReq.responseText;
+
+        do {
+            var unprocessed = messages.substring(nextReadPosition);
+            var messageEndIndex = unprocessed.indexOf("}\n\n");
+
+            if (messageEndIndex != -1)
+            {
+                var endOfMessageIndex = messageEndIndex + 3; // 3 is the length of "}\n\n"
+                var message = unprocessed.substring(0, endOfMessageIndex);
+                parse_message(message);
+                nextReadPosition += endOfMessageIndex;
+            }
+        } while (messageEndIndex != -1);
+
+        if (xhReq.readyState == 4)
+        {
+            $('#loading_indicator').css('display', 'none');
+            reset_polling();
+        }
+    }
+
+    function parse_message(messageJSON)
+    {
+        $('#loading_indicator').css('display', 'none');
+
+        messageJSON = messageJSON.trim();
+        var responseObject = JSON.parse(messageJSON);
+
+        // Parse object
+        if (responseObject.hasOwnProperty('errors'))
+        {
+            add_message('error', responseObject.errors)
+        }
+
+        if (responseObject.hasOwnProperty('warnings'))
+        {
+            add_message('warning', responseObject.warnings)
+        }
+
+        if (responseObject.hasOwnProperty('logs'))
+        {
+            add_message('log', responseObject.logs);
+        }
+
+        if (responseObject.hasOwnProperty('form'))
+        {
+            add_form(responseObject.form);
+        }
+    }
+
+    function add_message(type, messages)
+    {
+        // Get message containers
+        var errorContainer = $('#error-container');
+        var warningContainer = $('#warning-container');
+        var logContainer = $('#log-container');
```

Comment:
These variables should all have dollar signs. I won't comment on any of the other ones